crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -71,44 +77,75 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                 }
                 write!(f, [first.format()])?;
             }
-            SuiteKind::Class if is_docstring(first) => {
-                if !comments.has_leading_comments(first) && lines_before(first.start(), source) > 1
-                {
-                    // Allow up to one empty line before a class docstring, e.g., this is
-                    // stable formatting:
+            SuiteKind::Function => {
+                if let Some(constant) = get_docstring(first, &f.context().locator()) {
+                    write!(
+                        f,
+                        [
+                            // We format the expression, but the statement carries the comments
+                            leading_comments(comments.leading_comments(first)),
+                            constant
+                                .format()
+                                .with_options(ExprConstantLayout::String(StringLayout::DocString)),
+                            trailing_comments(comments.trailing_comments(first)),
+                        ]
+                    )?;
+                } else {
+                    write!(f, [first.format()])?;
+                }
+            }
+            SuiteKind::Class => {
+                if let Some(constant) = get_docstring(first, &f.context().locator()) {
+                    if !comments.has_leading_comments(first)
+                        && lines_before(first.start(), source) > 1
+                    {
+                        // Allow up to one empty line before a class docstring
+                        // ```python
+                        // class Test:
+                        //
+                        //     """Docstring"""
+                        // ```
+                        write!(f, [empty_line()])?;
+                    }
+                    write!(
+                        f,
+                        [
+                            // We format the expression, but the statement carries the comments
+                            leading_comments(comments.leading_comments(first)),
+                            constant
+                                .format()
+                                .with_options(ExprConstantLayout::String(StringLayout::DocString)),
```

Comment:
Calling the expression directly will be problematic when we add `fmt: skip` support because it now becomes necessary to handle the suppression comments here as well.

Reply:
i'm happy to change to whatever makes it easier with suppression. I initially had a `format_docstring` call here but thought it would be marginally nicer to at least go through the ExprConstant